hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -305,10 +308,14 @@ coverFee_ pparams systemStart epochInfo lookupUTxO walletUTxO partialTx@Babbage.
     Just (i, o) ->
       Right (i, o)
 
-  -- TODO: Do a better fee estimation based on the transaction's content.
-  calculateNeedlesslyHighFee (Redeemers redeemers) =
-    let executionCost = txscriptfee (_prices pparams) $ foldMap snd redeemers
-     in Coin 2_000_000 <> executionCost
+  calculatedFee = Coin calculated
```

Comment:
Can you do this? Are `Coin` and `Lovelace` represented using the same precision?